js/service/shell/index.js

```diff
@@ -44,3 +44,20 @@ exports.setCommand = function(name, cb) {
   assert(typeutils.isFunction(cb));
   commands.set(name, cb);
 };
+
+exports.runCommand = function(name, args, cb) {
+  var capture = require('./captureio.js');
+
+  assert(typeutils.isString(name));
+
+  args = args || "";
+
+  commands.get(name)(args, capture.io, function() { capture.events.emit('end'); });
+
+  if (cb) {
+    capture.events.redrawPrompt = inputBox.done;
+    cb(capture.events);
+  }
```

Comment:
I don't really understand what this does and why it's needed? I like the node-convention `cb(err[, args1, args2...)` but here it's called as the first argument? Also, I'd expect a callback to be asynchronous.